### PR TITLE
bug 1116335 - utf8_distinct_ci collation for Travis CI MySQL

### PR DESCRIPTION
**Pull request by groovecoder**
Just sending a PR to trigger a Travis build ...

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
         'ATOMIC_REQUESTS': True,
         'TEST': {
             'CHARSET': 'utf8',
-            'COLLATION': 'utf8_general_ci',
+            'COLLATION': 'utf8_distinct_ci',
```
Comment (jezdez) [inline settings.py]: Why is this needed? The test runner will run the migrations and therefore apply the collation to the tables needing it.

File: scripts/travis-before-script.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import codecs
```
Comment (jezdez) [inline scripts/travis-before-script.py]: Ough man, that's janky. I'm r- on this.

File: tox.ini [hunk 3]
```diff
@@ -19,7 +19,7 @@ setenv =
 deps=
     -rrequirements/compiled.txt
 commands =
-    coverage run manage.py test --noinput -v2 {posargs:kuma}
+    sudo coverage run manage.py test --noinput -v2 {posargs:kuma}
```
Comment (jezdez) [inline tox.ini]: Why sudo?

**Comment by groovecoder** [comment]
For all of these comments: I'm just trying things out to see what the best way to make Travis work is.

File: settings.py (hunk 1 above)
Comment (groovecoder) [inline settings.py]: The test runner fails to apply the collation migration with a MySQL `OperationalError` so I'm trying some other approaches.

File: scripts/travis-before-script.py (hunk 2 above)
Comment (groovecoder) [inline scripts/travis-before-script.py]: `codecs` is janky?

File: settings.py (hunk 1 above)
Comment (jezdez) [inline settings.py]: It didn't for me, is this in your VM? What's the traceback?

File: scripts/travis-before-script.py (hunk 2 above)
Comment (jezdez) [inline scripts/travis-before-script.py]: The whole file is :(

File: settings.py (hunk 1 above)
Comment (groovecoder) [inline settings.py]: In the Travis CI environment. I'm still poking at it trying things today

File: scripts/travis-before-script.py (hunk 2 above)
Comment (groovecoder) [inline scripts/travis-before-script.py]: It's not my favorite. On the plus side, it makes use of the Jinja templates we have for Ansible. But, it adds another discrepancy between Travis CI and dev environments. I was searching for a way to running ansible on Travis CI, but all I could find were guides to test ansible **roles** with Travis.

Though I recall we might be able to make an inventory file just for Travis with a single "localhost" entry? I'll try that too.

**Comment by groovecoder** [comment]
Closing. https://github.com/mozilla/kuma/commit/ed67af1a55b5a4ce937d1688ff875899277edc84 and https://github.com/mozilla/kuma/commit/5c89cc2b00dc13220c909a013682f9ebb32de712 are the better code to do this.
